ui/src/Update.elm

```diff
@@ -21,29 +21,43 @@ navigateToPage ({ filters } as model) newPage =
             ! [ newUrl <| urlFromRoute updatedRoute ]
 
 
-updateFilters : NewFilter -> Filters -> Filters
-updateFilters newFilter filters =
-    case newFilter of
-        ClearAll ->
-            Init.initFilters
+updateFilters : Facets -> NewFilter -> Filters -> Filters
+updateFilters facets newFilter filters =
+    let
+        toggleFilter facets value values =
+            let
+                newValues =
+                    if List.member value values then
+                        List.filter (\v -> v /= value) values
+                    else
+                        value :: values
+            in
+                if List.length newValues == List.length facets then
```

Comment:
This is to clear all of them if they all get checked? Maybe a comment would be helpful?